File: 645-set-mismatch/set-mismatch.py
class Solution(object):
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        
        # s = set()
        # duplicate = 0
        # for n in nums:
        #     if n in s:
        #         duplicate = n
        #     s.add(n)
        # for i in range(len(nums)):
        #     if i+1 not in s:
        #         return [duplicate, i+1]

        n, a, b = len(nums), sum(nums), sum(set(nums))
		
        s = n*(n+1)//2
        
        return [a-b, s-b]

File: 645-set-mismatch/set-mismatch.java
import java.util.*;

class Solution {
    public int[] findErrorNums(int[] nums) {
        int n = nums.length;
        int a = 0, b = 0;
        
        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            a += num;
            set.add(num);
        }
        for (int num : set) {
            b += num;
        }
        
        int s = n * (n + 1) / 2;
        
        return new int[]{a - b, s - b};
    }
}

